### PR TITLE
Fix for potential race condition in Mqtt protocol handler (gh#223)

### PR DESCRIPTION
**Pull request by ewertons**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [x] This pull-request is submitted against the `master` branch.

## Description of the changes

Fix for potential race condition in MqttIotHubAdapter.cs

## Reference/Link to the issue solved with this PR (if any)

https://github.com/Azure/azure-iot-sdk-csharp/issues/223

### CODE REVIEW

File: iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs [hunk 1]
```diff
@@ -271,7 +269,7 @@ public override void UserEventTriggered(IChannelHandlerContext context, object @
         #endregion
 
         #region Connect
-        async void Connect(IChannelHandlerContext context)
+        async Task Connect(IChannelHandlerContext context)
         {
             if (Logging.IsEnabled) Logging.Enter(this, context.Name, nameof(Connect));
```
Comment (prmathur-microsoft) [inline iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs]: Is it possible to capture this in E2E ?

Comment (ewertons) [reply iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs]: That is already captured in the existing E2E tests. All the mqtt-related E2E tests go through this code path.

**Pull request merged**